internal/scheduler/jobdb/job.go

```diff
@@ -604,7 +604,11 @@ func (job *Job) ActiveRunTimestamp() int64 {
 
 // InTerminalState returns true if the job  is in a terminal state
 func (job *Job) InTerminalState() bool {
-	return job.succeeded || job.cancelled || job.failed
+	isRunInTerminal := false
+	if latestRun := job.LatestRun(); latestRun != nil {
+		isRunInTerminal = latestRun.Succeeded() || latestRun.Cancelled() || latestRun.Failed() || latestRun.Preempted()
```

Comment:
I thought that if we have the latest run reached a terminal state, we should also report the job as in terminal state and return `true`.